migration/migrator/migrations/course/20210111163413_add_poll_data_folder.py

```diff
@@ -0,0 +1,23 @@
+import os
+import grp
+from pathlib import Path
+
+def up(config, database, semester, course):
+    course_dir = Path(config.submitty['submitty_data_dir'], 'courses', semester, course, 'reports')
+    polls_dir = Path(course_dir, 'polls')
+
+    #create directory
+    polls_dir.mkdir(mode=0o750, exist_ok=True)
+
+    php_user = config.submitty_users['php_user']
+
+    # get course group
+    course_group_id = course_dir.stat().st_gid
+    course_group = grp.getgrgid(course_group_id)[0]
+
+    # set the owner/group/permissions
+    os.system("chown -R "+php_user+":"+course_group+" "+str(polls_dir))
```

Comment:
We want to use built-in python stuff as much as possible so that we get error handling and such if things blow up. By using `os.system`, we would shallow any possible errors that might happen there. So for chown, we can use [`shutil.chown`](https://docs.python.org/3/library/shutil.html#shutil.chown). As this then relies on `os.chown`, we can also use the `Path` directly without having to first convert it to a string.

```suggestion
    shutil.chown(polls_dir, php_user, course_group_id)
```